pallets/rmrk-core/src/lib.rs

```diff
@@ -83,19 +77,28 @@ pub mod pallet {
 			+ Into<Self::InstanceId>;
 
 		type ResourceId: Member + Parameter + Default + Copy + HasCompact + AtLeast32BitUnsigned;
-	}
 
-	/// Next available collection ID.
-	#[pallet::storage]
-	#[pallet::getter(fn next_collection_id)]
-	pub type NextCollectionId<T: Config> = StorageValue<_, T::CollectionId, ValueQuery>;
+		/// The auction ID type.
+		type CollectionId: Parameter
```

Comment:
Is it necessary to allow arbitrary CollectionId? Substrate was criticized by over-design of allowing almost the types of its pallets to be customizable, and in the end they have created smoldot to make it more lightweight